hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -2080,7 +2075,7 @@ public long createTable(final TableDescriptor tableDescriptor, final byte[][] sp
     this.clusterSchemaService.getNamespace(namespace);
 
     RegionInfo[] newRegions = ModifyRegionUtils.createRegionInfos(desc, splitKeys);
-    sanityCheckTableDescriptor(desc);
+    TableDescriptorChecker.sanityCheck(conf, desc, masterCheckCompression, masterCheckEncryption);
```

Comment:
Can just pass the conf to the sanityCheck ?  the masterCheckCompression & masterCheckEncryption argument don't need anymore... also the two member in HMaster class can be removed now ?